src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -0,0 +1,430 @@
+                                                    final boolean allowMissingData,
+                                                    final boolean splitMultiAllelic) {
+
+        final int numRecordsToProduce = splitMultiAllelic ? vc.getAlternateAlleles().size() : 1;
+        final List<List<String>> records = new ArrayList<>(numRecordsToProduce);
+
+        int numFields = fields.size();
+        final boolean addGenotypeFields = genotypeFields != null && !genotypeFields.isEmpty();
+        if ( addGenotypeFields ) {
+            numFields += genotypeFields.size() * samples.size();
+        }
+
+        for ( int i = 0; i < numRecordsToProduce; i++ ) {
+            records.add(new ArrayList<>(numFields));
+        }
+
+        for ( final String field : fields ) {
+            if ( splitMultiAllelic && field.equals("ALT") ) { // we need to special case the ALT field when splitting out multi-allelic records
+                addFieldValue(splitAltAlleles(vc), records);
+            } else if ( getters.containsKey(field) ) {
+                addFieldValue(getters.get(field).apply(vc), records);
+            } else if ( vc.hasAttribute(field) ) {
+                addFieldValue(vc.getAttribute(field, null), records);
+            } else if ( isWildCard(field) ) {
+                final Set<String> wildVals = new HashSet<>();
+                for ( final Map.Entry<String,Object> elt : vc.getAttributes().entrySet()) {
+                    if ( elt.getKey().startsWith(field.substring(0, field.length() - 1)) ) {
+                        wildVals.add(elt.getValue().toString());
+                    }
+                }
+
+                String val = MISSING_DATA;
+                if (!wildVals.isEmpty()) {
+                    final List<String> toVal = new ArrayList<>(wildVals);
+                    Collections.sort(toVal);
+                    val = Utils.join(",", toVal);
+                }
+
+                addFieldValue(val, records);
+            } else if ( ! allowMissingData ) {
+                throw new UserException(String.format("Missing field %s in vc %s at %s", field, vc.getSource(), vc));
+            } else {
+                addFieldValue(MISSING_DATA, records);
+            }
+        }
+
+        if ( addGenotypeFields ) {
+            for ( final String sample : samples ) {
+                for ( final String gf : genotypeFields ) {
+                    if ( vc.hasGenotype(sample) && vc.getGenotype(sample).hasAnyAttribute(gf) ) {
+                        if ( gf.equals(VCFConstants.GENOTYPE_KEY) ) {
+                            addFieldValue(vc.getGenotype(sample).getGenotypeString(true), records);
+                        } else {
+                            addFieldValue(vc.getGenotype(sample).getAnyAttribute(gf), records);
+                        }
+                    }
+                    else {
+                        addFieldValue(MISSING_DATA, records);
```

Comment:
It doesn't look like the genotype fields honor the behavior for the allowMissingData flag.